MovieApp/MovieApp/App/DataRepository/MoviesDataRepository.swift

```diff
@@ -0,0 +1,43 @@
+class MoviesDataRepository: MoviesDataRepositoryProtocol {
+    
+    static let shared: MoviesDataRepositoryProtocol = MoviesDataRepository()
+    
+    private let networkDataSource: MovieNetworkDataSourceProtocol!
+    
+    init() {
+        self.networkDataSource = MovieNetworkDataSource.shared
+    }
+    
+    func fetchPopularMovies(completion: @escaping (Result<[MovieDataRepositoryModel]?, RequestError>) -> Void) {
+        networkDataSource.fetchPopularMovies { result in
+            switch result {
+            case .success(let movies):
+                let mappedMovies = self.mapMovies(movies)
+                completion(.success(mappedMovies))
+            case .failure(let error):
+                completion(.failure(error))
+            }
+        }
+    }
+    
+}
+
+extension MoviesDataRepository {
+    
+    private func mapMovies(_ movies: [MovieNetworkDataSourceModel]?) -> [MovieDataRepositoryModel]?{
+        return movies?.map {
+            return MovieDataRepositoryModel(
```

Comment:
If there's no logic to it, models can map themselves, but only the upper layer model can know about the lower level model, not the other way around.

For example, `MovieDataRepositoryModel` can have:
1. `init` that receives `MovieNetworkDataSourceModel`
2. `toModel()` function that returns `MovieNetworkDataSourceModel`

So the repo model can init itself with network data, and map itself to network model if used in a network request. There's no need for both functions all the time.

Reply:
Added inits to all models except MovieResponse.
[MovieVM](https://github.com/lihter/Movie-App/pull/24/files#diff-a3ee473b9e1ed2108bce116a20cb459775b95f14a6c7a496919a78d3cab3f4a6R14)
[MovieDataModel](https://github.com/lihter/Movie-App/pull/24/files#diff-8619b2817b5d842b8b373e65db34f09b43f9568b4cbdc658cde0f60e682db4f0R18)
[MovieRepoModel](https://github.com/lihter/Movie-App/pull/24/files#diff-16ec28a5469846b8ad26272a1925c9f70fa4bf8572c2a83c4b764c9caa874313R20)
[MovieModel](https://github.com/lihter/Movie-App/pull/24/files#diff-b2ce96a4be225b4224eac0473fa85c4006296a2c071734276d6fa283d8777cb7R20)